Duplicate automation step [STOMAIL-7459]

pavel-mailpoet opened the pull request:
## Description

1. Add new Duplicate step item to step menu (behind the three dots)
2. Don't allow duplicating if/else step, or triggers
3. When duplicating Send email step, also duplicate the email assigned. 

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/966

## Linked tickets

[STOMAIL-7459]

## After-merge notes

_N/A_

## Tasks

- [x] I have added a changelog entry (`./do changelog:add --type=<type> --description=<description>`)
- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:stomail-7459-duplicate-automation-step)

_The latest successful build from `stomail-7459-duplicate-automation-step` will be used. If none is available, the link won't work._

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a "Duplicate step" option in the automation editor's step menu for eligible steps, with a modal explaining it is a premium feature.
  * Enabled duplication of automation steps, including proper handling of duplicated newsletter steps to create copies with updated details.
  * Improved the "Edit content" button for duplicated email steps to handle backend duplication and redirect correctly.

* **Bug Fixes**
  * None.

* **Tests**
  * Added integration and unit tests to ensure correct duplication behavior for newsletter steps and step duplication logic.

* **Documentation**
  * Updated the changelog to document the addition of the step duplication feature.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

pavel-mailpoet committed:
Add duplicate step menu item

[STOMAIL-7459]

pavel-mailpoet committed:
Add free logic for duplication menu item

[STOMAIL-7459]

pavel-mailpoet committed:
Hide duplication on trigger steps

[STOMAIL-7459]

pavel-mailpoet committed:
Remove TODO comment

[STOMAIL-7459]

pavel-mailpoet committed:
Add changelog

pavel-mailpoet committed:
Improve the duplication component

pavel-mailpoet committed:
Add duplication logic for the send email step

[STOMAIL-7459]

pavel-mailpoet committed:
Add a test

[STOMAIL-7459]

pavel-mailpoet committed:
Fix code review issues

[STOMAIL-7459]

pavel-mailpoet committed:
Fix automation step args not being updated in PUT endpoint

pavel-mailpoet committed:
Refactor code to make it more readable

NeosinneR reviewed (changes requested):
Thanks for changes, @pavel-mailpoet. The code looks and works as expected, and I think the interface update improves extensibility. 

I found an issue when duplicating email, which causes that the original instead of duplicated email is edited, while from the UI, it seems like you're editing the duplicated. Here are steps to reproduce, or a video below to see the issue.
1. Create a `Send email` step with an email.
2. Duplicate it and click on the duplicated one to open the step settings.
3. Save the automation, and click `Edit content` on the email.

This will cause you to edit the original email, because even though when saving the automation the duplicated step is correctly assigned a new duplicated email, step settings UI is not updated with the new ID. Only when you again click on the duplicated step, it will re-render the step settings with the updated email ID. 

https://github.com/user-attachments/assets/61ee5a2d-d388-417d-8c0c-ed70166f0afc

NeosinneR committed:
Fix indent

STOMAIL-7459

pavel-mailpoet committed:
Fix email editing for duplicated automation steps

When duplicating a “Send email” step, users were editing the original email
instead of the duplicated one. This happened because the email duplication
only occurred on backend save, but the UI was using the original email ID.

[STOMAIL-7459]

NeosinneR reviewed (approved):
Thanks for the changes, @pavel-mailpoet. Looks and works as expected.

Aschepikov reviewed (approved):
GJ!
